packages/cozy-harvest-lib/src/helpers/konnectors.js

```diff
@@ -103,6 +109,130 @@ export const needsFolder = konnector => {
+ * corresponding localized name.
+ * @return {String}         Localized directory
+ */
+const renderBaseDir = (baseDir, folders = {}) => {
+  // Look for variable name into folders but without $ prefix
+  const renderedBaseDir =
+    folders[baseDir.slice(1)] || DEFAULT_LOCALIZED_BASE_DIR
+  // Trim `/` and avoid multiple `/` characters with regexp
+  return trim(renderedBaseDir.replace(/(\/+)/g, '/'), '/')
+}
+
+/**
+ * Render the given folder path using the given `variables` object.
+ * Available variable are `$konnector` (konnector name) and `$account`
+ * (account label, i.e. id or name)
+ * @param  {String} path      Path to render
+ * @param  {Object} variables Object mapping variable to actual values
+ * @return {String}           Rendered path
+ */
+const renderSubDir = (path, variables = {}) => {
+  // Trim `/` and avoid multiple `/` characters with regexp
+  const sanitizedPath = trim(path.replace(/(\/+)/g, '/'), '/')
+
+  // Let's get only full variable name limited by '/'. We want to avoid false
+  // positive like parsing `$variableInString` to `valueInString`
+  const segments = sanitizedPath.split('/')
+  return segments
+    .map(segment => variables[segment.slice(1)] || segment)
+    .join('/')
+}
+
+/**
+ * Check if the provided Path start withs our allowedBaseDirPath to see
+ * @param {String} path
+ * @return {Boolean}
+ */
+const hasBaseDir = path => {
+  return allowedBaseDirVariables.some(baseDirVar => {
+    return path.startsWith(baseDirVar)
+  })
+}
+/**
+ * This method creates the subDir. We can't have an empty subDir, so we set
+ * it to our default '$konnector/$account'
+ * @param {String} fullPath String containing potentially the defaultDir
+ * @param {String} defaultDir String to remove from the fullPath
+ */
+const buildSubDir = (fullPath, defaultDir) => {
+  let buildedSubDir = fullPath.substring(defaultDir.length)
+  if (buildedSubDir === '') {
+    buildedSubDir = '$konnector/$account'
+  }
+  return buildedSubDir
+}
+/**
+ * Build folder path for a given konnector and a given account.
+ *
+ * If konnector.folders[0].defaultDir exists, it is used as default directory.
+ *
+ * Occurrences of following strings in base directory are replaced by:
+ * * `$administrative`: Administrative folder
+ * * `$photos`: Photos folder
+ *
+ * Occurrences of following strings in path are replaced by:
+ * * `$account: Account label (id or name)`
+ * * `$konnector`: Konnector name
+ *
+ * If no konnectors.folders[0].defaultDir is set, the default dir used is
+ * *  `$administrative/$konnector/$account`
+ *
+ * @param  {Object} konnector Konnector document
+ * @param  {Object} account   Account document
+ * @param  {Object} folders   Object containing a mapping from folder
+ * identifiers (ex: $administrative) to their localized values (ex:
+ * Administratif).
+ * @return {String}           The result path
+ */
+
+export const buildFolderPath = (konnector, account, folders = {}) => {
```

Comment:
I was expecting something simplier, like (not tested) : 

```javascript
const path = get(konnector, 'folders[0].defaultDir', '/$administrative/$konnector/$account')
const fullPath = (path[0]=='$') ? `/${path}` : path
const replace = { ...folders,  konnector: konnector.name, account: accounts.getLabel(account))
const op = function (txt, keys) {
  const [key, ...rest] = keys
  const res = txt.replace('$' + key, replace[key])
  return (rest.length > 0) ? op(res, rest) : res
}
return op(fullPath)
```
It has not all the complex behaviour that you had, but do we need them ? In my understanding, we start from zero or near zero (only two connectors using this kind of property, not live).

Reply:
Ça peut fonctionner mais ça veut dire qu'il faut qu'on soit vachement vigilant sur ce chemin dans le manifest. 

Car là, ça ne va pas vérifier qu'on commence avec les bons répertoires, que si on met juste `$photos` ça ne va pas aller automatiquement dans `Photos/konnecteur/account` mais juste dans `Photos` et donc potentiellement mettre un sacré binz. 

Au final, même si on a la main sur les connecteurs, ça me parait léger de compter sur le fait que notre attention sera toujours au top lors des merges sur ce point et donc j'ai la préférence pour du code défensif quitte à le complexifier (un peu). 

WDYT?

Reply:
Ok. On a le code complexe déjà fait, tu penses qu'il est utile, laissons le. Je n'ai pas de raison de bloquer.